core/src/main/java/com/amplitude/core/utilities/JSONUtil.kt

```diff
@@ -76,7 +76,7 @@ object JSONUtil {
         return eventsArray.toString()
     }
 
-    private fun truncate(obj: JSONObject?): JSONObject? {
+    private fun truncate(obj: JSONObject?): JSONObject {
```

Comment:
Why was this updated? I feel in some cases we want to have nil for empty fields we don't want to transfer empty fields.

I looked at the checks and noticed new warnings in the build from this update. Can you please look into the warnings and make sure this won't cause any problems.

<img width="1102" alt="image" src="https://github.com/amplitude/Amplitude-Kotlin/assets/5790872/8c79863a-7854-4d5e-8455-5c9f24ffdc65">

Last build before changes:
https://github.com/amplitude/Amplitude-Kotlin/actions/runs/4991994545/jobs/8939218744

After:
https://github.com/amplitude/Amplitude-Kotlin/actions/runs/4961668962/jobs/8878750018

Reply:
> Why was this updated? I feel in some cases we want to have nil for empty fields we don't want to transfer empty fields.

Android Studio shows warning:
![image](https://github.com/amplitude/Amplitude-Kotlin/assets/9509929/2d46e1ca-ad1e-40f3-862d-c5fe65befc30)

Reply:
> I looked at the checks and noticed new warnings in the build from this update. Can you please look into the warnings and make sure this won't cause any problems.

I tried locally this branch and `main` - both have the same build warnings after `clean`:
```
./gradlew clean
./gradlew build
```

I can fix some of them that look safe.

Reply:
Fixed some "simple" warnings

Reply:
Thanks @falconandy. The build output looks much better 🙏